Sources/Overlays/_Testing_WinSDK/Attachments/_AttachableImageWrapper.swift

```diff
@@ -19,91 +18,133 @@ internal import WinSDK
+  let imageFormat: AttachableImageFormat?
 
-  /// Whether or not to call `_cleanUpAttachment(at:)` on `pointer` when this
+  /// Whether or not to call `_deinitializeAttachableValue()` on `image` when this
   /// instance is deinitialized.
   ///
-  /// - Note: If cleanup is not performed, `pointer` is effectively being
-  ///   borrowed from the calling context.
-  var cleanUpWhenDone: Bool
-
-  init(image: Image, imageFormat: AttachableImageFormat?, cleanUpWhenDone: Bool) {
-    self.image = image
+  /// - Note: If deinitialization is not performed and `image` is a type that
+  ///   does not participate in ARC, `image` is effectively being borrowed from
+  ///   the calling context.
+  private let _deinitializeWhenDone: Bool
+
+  init(image: borrowing Image, imageFormat: AttachableImageFormat?, deinitializeWhenDone: Bool) {
+    self.image = Result { [image = copy image] in
+      try image._copyAttachableValue()
+    }
     self.imageFormat = imageFormat
-    self.cleanUpWhenDone = cleanUpWhenDone
+    self._deinitializeWhenDone = deinitializeWhenDone
   }
 
   deinit {
-    if cleanUpWhenDone {
-      image._cleanUpAttachment()
+    if _deinitializeWhenDone, let image = try? image.get() {
+      image._deinitializeAttachableValue()
     }
   }
 }
 
-@available(*, unavailable)
-extension _AttachableImageWrapper: Sendable {}
-
 // MARK: -
 
 extension _AttachableImageWrapper: AttachableWrapper {
-  public var wrappedValue: Image {
-    image
+  public var wrappedValue: Image? {
+    try? image.get()
   }
 
-  public func withUnsafeBytes<R>(for attachment: borrowing Attachment<Self>, _ body: (UnsafeRawBufferPointer) throws -> R) throws -> R {
+  public func withUnsafeBytes<R>(for attachment: borrowing Attachment<_AttachableImageWrapper>, _ body: (UnsafeRawBufferPointer) throws -> R) throws -> R {
     // Create an in-memory stream to write the image data to. Note that Windows
     // documentation recommends SHCreateMemStream() instead, but that function
     // does not provide a mechanism to access the underlying memory directly.
     var stream: UnsafeMutablePointer<IStream>?
     let rCreateStream = CreateStreamOnHGlobal(nil, true, &stream)
     guard S_OK == rCreateStream, let stream else {
-      throw GDIPlusError.streamCreationFailed(rCreateStream)
+      throw ImageAttachmentError.comObjectCreationFailed(IStream.self, rCreateStream)
     }
     defer {
-      stream.withMemoryRebound(to: IUnknown.self, capacity: 1) { stream in
-        _ = swt_IUnknown_Release(stream)
-      }
+      _ = stream.pointee.lpVtbl.pointee.Release(stream)
     }
 
-    try withGDIPlus {
-      // Get a GDI+ image from the attachment.
-      let image = try image._copyAttachableGDIPlusImage()
-      defer {
-        swt_GdiplusImageDelete(image)
-      }
+    // Get an imaging factory to create the WIC bitmap and encoder.
+    let factory = try IWICImagingFactory.create()
+    defer {
+      _ = factory.pointee.lpVtbl.pointee.Release(factory)
+    }
```

Comment:
I wonder if it makes sense to have a global cached value for the factory at the module level.

Reply:
I considered it, but the documentation doesn't say it's thread-safe, so I erred on the side of caution.